### PR TITLE
 python 3.6 support removed due to EOL, 2021-12-23

### LINKED ISSUE #2363: Remove python 3.6 support due to EOL, 2021-12-23
Let's remove python 3.6 from our CI from due to EOL, 2021-12-23 (https://www.python.org/downloads/)

### PR DESCRIPTION
**Pull request by bibhabasumohapatra**
Fixes #2363

Description: Let's remove python 3.6 from our CI from due to EOL, 2021-12-23 (https://www.python.org/downloads/)

hey @vfdev-5 ,
 to remove python-version 3.6 and py36 support from
IG I can remove that the following way from these paths ,right? 

1. **.github/workflows/unit-tests.yml**

2. **.github/workflows/pytorch-version-tests.yml**

 `python-version: [3.6, 3.7, 3.8, 3.9]`    ->   `python-version: [3.7, 3.8, 3.9] `

3. **pyproject.toml**
 `target-version = ['py36', 'py37', 'py38']`  -> `target-version = [ 'py37', 'py38']` 

Check list:

- [ ] removing support for python 3.6 from CI

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [tool.black]
 line-length = 120
-target-version = ['py36', 'py37', 'py38']
+target-version = ['py37', 'py38']
```
Comment (vfdev-5) [inline pyproject.toml:3]: Can we add `py39` here, please

Comment (bibhabasumohapatra) [reply pyproject.toml:3]: okay! will do it.

File: pyproject.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [tool.black]
 line-length = 120
-target-version = ['py37', 'py38']
+target-version = ['py37', 'py38','py39']
```
Comment (vfdev-5) [inline pyproject.toml:3]:
```suggestion
target-version = ['py37', 'py38', 'py39']
```
please keep the same formatting with spaces.

File: .github/workflows/pytorch-version-tests.yml [hunk 3]
```diff
@@ -13,7 +13,7 @@ jobs:
       max-parallel: 10
       fail-fast: false
       matrix:
-        python-version: [3.6, 3.7, 3.8]
+        python-version: [3.7, 3.8]
```
Comment (sdesrozis) [inline .github/workflows/pytorch-version-tests.yml:16]: Should we add python 3.9 here ?

File: pyproject.toml (hunk 2 above)
Comment (vfdev-5) [reply pyproject.toml]: This is unfortunately not possible with fixed version of black we are using

File: .github/workflows/pytorch-version-tests.yml (hunk 3 above)
Comment (vfdev-5) [reply .github/workflows/pytorch-version-tests.yml:16]: We should then exclude older pytorch version that were not delivered for python 3.9

Comment (bibhabasumohapatra) [reply .github/workflows/pytorch-version-tests.yml:16]: @vfdev-5 should we add python 3.9 and exclude PyTorch old version?

Comment (vfdev-5) [reply .github/workflows/pytorch-version-tests.yml:16]: @bibhabasumohapatra let's try that

Comment (bibhabasumohapatra) [reply .github/workflows/pytorch-version-tests.yml:16]: @vfdev-5  no need for PyTorch and 3.9 update in this PR right?

Comment (trsvchn) [reply .github/workflows/pytorch-version-tests.yml:16]: > We should then exclude older pytorch version that were not delivered for python 3.9

yes like we did here:
https://github.com/pytorch/ignite/blob/94bf2dab857f86e3045237013d94a8bde4e164cf/.github/workflows/pytorch-version-tests.yml#L18-L20

we just need to know the exact versions

Comment (vfdev-5) [reply .github/workflows/pytorch-version-tests.yml:16]: @bibhabasumohapatra we can include that in this PR

**Comment by sdesrozis** [comment]
@bibhabasumohapatra All the tests failed. It seems related to https://github.com/pytorch/ignite/pull/2365#discussion_r771964202

Could you explore that ? Thanks !

**Comment by bibhabasumohapatra** [comment]
> @bibhabasumohapatra All the tests failed. It seems related to [#2365 (comment)](https://github.com/pytorch/ignite/pull/2365#discussion_r771964202)

Hey! @sdesrozis  I was asking that - it shows 
Error:  Invalid value for '-t' / '--target-version': invalid choice: py39. (choose from py27, py33, py34, py35, py36, py37, py38)
why is that?

**Comment by sdesrozis** [comment]
> > @bibhabasumohapatra All the tests failed. It seems related to [#2365 (comment)](https://github.com/pytorch/ignite/pull/2365#discussion_r771964202)
> 
> Hey! @sdesrozis I was asking that - it shows Error: Invalid value for '-t' / '--target-version': invalid choice: py39. (choose from py27, py33, py34, py35, py36, py37, py38) why is that?

I don't know. See [#2365 (comment)](https://github.com/pytorch/ignite/pull/2365#discussion_r771964202), you added `py39` in `pyproject.toml` and it seems to make it fail.

**Comment by bibhabasumohapatra** [comment]
@sdesrozis @vfdev-5  If I add py39 it fails, the version of black we are using only allows -
 `-t, --target-version [py27|py33|py34|py35|py36|py37|py38]` 
_checked this by using_ `black -h` _command_
**so if we add py39 fails all tests**. may updating black solve this problem.
Now I have checked the tests and all checks are passed without 'py39' 
image

**Comment by sdesrozis** [comment]
> @sdesrozis @vfdev-5 If I add py39 it fails, the version of black we are using only allows -
> `-t, --target-version [py27|py33|py34|py35|py36|py37|py38]`
> _checked this by using_ `black -h` _command_
> **so if we add py39 fails all tests**. may updating black solve this problem.
> Now I have checked the tests and all checks are passed without 'py39'
> image

Let’s see whether the CI works fine

**Review by sdesrozis — APPROVED** [review]
@bibhabasumohapatra Thank you, LGTM !

**Review by trsvchn — APPROVED** [review]
Thanks @bibhabasumohapatra for the PR! LGMT!

**Pull request merged**